### PR TITLE
[#2151] Update Typescript-related Major Dependencies

### LINKED ISSUE #2151: Frontend DevOps: Update Node.js dependencies
### What feature(s) would you like to see in RepoSense

Some of our frontend dependencies have been flagged as out of date:
image
image

### Additional context
This issue is a continuation of #1871.

### PR DESCRIPTION
**Pull request by sopa301**
Fixes #2151.

### Proposed commit message

```
Update Typescript-related Major Dependencies

Many of our TypeScript-related dependencies are outdated.

Let's update them.
```

### Other information

TypeScript isn't updated to 5.0.0 or later due to [a bug in a dependency](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/issues/838). However, it doesn't seem to be a major bug, so we can proceed with updating Typescript if it's not a big concern.

#2125 seems to trigger the new `indent` rule quite fiercely. I'm not sure why there is a disparity between the new and old `indent` rules. Perhaps we should discuss which behaviour we prefer?

### CODE REVIEW

**Review by jonasongg — APPROVED** [review]
LGTM! might be better to remove the changes to remove whitespace? i realise this pr was made before #2182 though, so apologies for not noticing this earlier!

File: frontend/.eslintrc.json
Comment (ckcherry23) [inline frontend/.eslintrc.json:37]: Any reason this was removed?

File: frontend/.eslintrc.json
Comment (ckcherry23) [inline frontend/.eslintrc.json:45]: Same for this? Maybe good to give a one-line reason for everything removed or newly added

File: frontend/.eslintrc.json
Comment (sopa301) [inline frontend/.eslintrc.json:49]: Added namespace.

File: frontend/.eslintrc.json
Comment (sopa301) [inline frontend/.eslintrc.json:77]: `SwitchCase` is 0 by default.

File: frontend/.eslintrc.json
Comment (sopa301) [inline frontend/.eslintrc.json:83]: Added namespace as part of the rules migration.

File: frontend/.eslintrc.json
Comment (sopa301) [inline frontend/.eslintrc.json:68]: `disable-legacy` had to be re-added here to disable `typescript-eslint/recommended` old linting rules from interfering with the new linting rules.

File: frontend/.eslintrc.json
Comment (sopa301) [inline frontend/.eslintrc.json:12]: Enable removed rules.

**Review by ckcherry23 — APPROVED** [review]
@sopa301 LGTM! Thanks for meticulously looking through all of the updates needed!

**Review by vvidday — APPROVED** [review]
LGTM, thanks

**Comment by ckcherry23** [comment]
@sopa301 Please fix the merge conflicts! We will merge the PR 2 days after the final approval (so tmrw).

**Pull request merged**
